src/amo/components/SearchForm/index.js

```diff
@@ -114,20 +114,32 @@ export class SearchFormBase extends React.Component {
     this.setState({ searchValue: event.target.value });
   }
 
-  handleSuggestionsFetchRequested = this.props.debounce(({ value, reason }) => {
+  dispatchAutocompleteStart = this.props.debounce(({ filters }) => {
+    const { dispatch, errorHandler } = this.props;
+
+    dispatch(autocompleteStart({
+      errorHandlerId: errorHandler.id,
+      filters,
+    }));
+    this.setState({ oldAutocompleteFilters: filters });
+  }, 200, { leading: true })
+
+  handleSuggestionsFetchRequested = ({ value, reason }) => {
     if (!value) {
       log.debug(oneLine`Ignoring suggestions fetch requested because value
         is not supplied: ${value}`);
       return;
     }
 
-    const { addonType, dispatch, errorHandler } = this.props;
+    const { addonType } = this.props;
     const filters = { query: value };
 
     if (addonType) {
       filters.addonType = addonType;
     }
 
+    this.setState({ autocompleteIsOpen: true });
```

Comment:
To make things a little less complex, you could just add the filter change to this and remove it from the dispatch helper:

```js
this.setState({ 
  autocompleteIsOpen: true, oldAutocompleteFilters: filters, 
});
```